app/main.py

```diff
@@ -1 +1,15 @@
-# write your code here
+def create_report(data_file_name: str, report_file_name: str):
+    supply = 0
+    buy = 0
+    with open(data_file_name, 'r') as filein:
+        for line in filein:
+            items = line.split(",")
+            if len(items) > 0 and items[0] == 'supply':
+                supply += int(items[1])
+            elif len(items) > 0 and items[0] == 'buy':
+                buy += int(items[1])
+    result = supply - buy
+    dict_result = {"supply": supply, "buy": buy, "result": result}
```

Comment:
Why do you use this dict? 
It looks like unnecessary code

Reply:
because DRY, ex:
```
with open(report_file_name, 'w') as fileout:
        for name, value in dict_result.items():
            fileout.writelines(f'{str(name)},' + str(value) + '\n')
```
```
with open(report_file_name, 'w') as fileout:
        fileout.writelines(value1 + str(value) + '\n')
        fileout.writelines(value2 + str(value2) + '\n')
        fileout.writelines(value3 + str(value3) + '\n')
```

Reply:
"Explicit is better than implicit" works in this case.
Trying to avoid code duplication you write two extra lines:
```
dict_result = {"supply": supply, "buy": buy, "result": result}
for name, value in dict_result.items():
```
So, you still have three lines of code, but they become much more complicated.

Reply:
I agree, thx for an explanation. I've fixed